src/config/mongoose_config_spec.erl

```diff
@@ -472,31 +467,29 @@ http_handler_required(_) -> [].
 
 %% path: (host_config[].)auth
 auth() ->
+    Items = maps:from_list([{a2b(Method), ejabberd_auth:config_spec(Method)} ||
+                               Method <- all_auth_methods()]),
```

Comment:
This is the only part that feels like a hack, and still, it's so much cleaner than before. Have modules do their own configs, yay! 🎉

Reply:
Yes, I know it's not that clean, but it's the same as for modules and services.